share/rocm/cmake/header_template.h.in

```diff
@@ -9,12 +9,16 @@
 /* include file */
 #include "@file_rel_path@"
 #else
+#if @deprecated_error@  // ROCM_HEADER_WRAPPER_WARN_AS_ERROR
```

Comment:
This will generate header file with 0 or 1 like below. 
#if 0  // ROCM_HEADER_WRAPPER_WARN_AS_ERROR
To enable/disable #error , i think a variable should be used , so that user can define the variable while building